### ISSUE TITLE
AD Group Related Operations

### ISSUE THREAD

**Issue by johnakash** [cannatag/ldap3#271]
Hi Team,

We are using the ldap3 python module for automating AD (Windows AD server) related activities (like unlock, useradd, adding a user to group, etc) from a RHEL 7 machine. 

We were successfylly performed user unlock and password reset using ldap3 module. Now we are looking in to the following processes. 

- adding a user to group
- listing groups associated with a user
- Removing a user from a group

We cannot find any options for performing above mentioned operations while using ldap3

![desctop screenshot](https://cloud.githubusercontent.com/assets/1567281/21176022/05a10f76-c20b-11e6-81c9-e9de233a5297.png)

Could you please let us know how can we achieve them (**adding a user to group, listing groups associated with a user and  Removing a user from a group**)?

**Comment by cannatag** [comment]
Hi, this are quite simple operation to perform. To add/remove a user to/from a group in AD just add/remove his dn to the 'member' attribute of the group. To know what users are members of a group just search for the 'member' attribute the group. This can be done with standard ldap operation and simple ldap3 code. Let me know if you find any problem. I could possibly include some functions in the extend.microsoft namespace in a next release of ldap3.

**Comment by johnakash** [comment]
Hi,
Thanks for the response, unfortunately I was not able to find it out (_the solution which you have mentioned_) and am not that much familiar with Python coding. Could you please let me know which documentation page I need to refer for this?

**Comment by cannatag** [comment]
OK. Will send some examples soon.

**Comment by cannatag** [comment]
To add user1_dn and user2_dn to group_dn:
```
conn.modify(group_dn, {'member': (MODIFY_ADD, [user_dn, user2_dn])}
```

To remove user1_dn and user2_dn from group_dn:
```
conn.modify(group_dn, {'member': (MODIFY_DELETE, [user_dn, user2_dn])}
```

To get all group members:
```
connection.search(group, '(objectclass=*)', BASE, attributes=['member'])
```

``MODIFY_ADD``, ``MODIFY_DELETE`` and ``BASE`` must be imported from the ``ldap3`` namespace.

Also in version 2.2 I have included two methods in the extend namespace:
```
conn.extend.microsoft.add_members_to_groups([user1, user2], [group1, group2])
conn.extend.microsoft.remove_members_from_groups([user1, user2], [group1, group2])
```

to add and remove users from groups.

version 2.2 is in dev. Will be released soon.

**Comment by cannatag** [comment]
included in v2.2.0
